arch/arm64/core/mmu.c

```diff
@@ -806,6 +806,79 @@ static uint64_t get_tcr(int el)
 	return tcr;
 }
 
+#ifdef CONFIG_DT_HAS_XEN_XEN_ENABLED
+
+static int flush_inv_mapped_pages(int (*cmo_fn)(void *, size_t),
+				  uint64_t *pte, int level, uint64_t base)
+{
+	int ret = 0;
+	uint64_t level_size = BIT(LEVEL_TO_VA_SIZE_SHIFT(level));
+	const uint64_t max_va_addr = (uint64_t)CONFIG_KERNEL_VM_BASE + CONFIG_KERNEL_VM_SIZE;
+
+	if (!xlat_use_count[table_index(pte)]) {
+		return 0;
+	}
+
+	for (unsigned int i = 0; i < Ln_XLAT_NUM_ENTRIES; i++) {
+		uint64_t end, va = base + i * level_size;
```

Comment:
`uintptr_r`

Reply:
Well, this is Zephyr, not U-Boot. It is expected that new code is aligned with Zephyr style.

Reply:
Could you please show me exact rule that isage of u64 ia restricted and gove a detailed explanation why it is prohibited?

Reply:
It is not exactly prohibited per se. But:

1. `uintptr_t` is guaranteed to represent any pointer value. So it is natural to use it for virtual addresses.
2. In MMU-related code it is easy to mix physical and virtual addresses. So it customary to use different types for them. For example, OP-TEE has distinct `paddr_t` and `vaddr_t`. Xen goes even further when it is working with GFNs and PFNs, so it introduces wrapper types `gfn_t` and `pfn_t` in a such way, that you can't assign GFN to PFN, as this is completely different types. As I see, Zephyr devs use `uintptr_t` for both `phys` and `virt` addresses and `uint64_t` for table entries. But they do not use `uint64_t` to store VAs.

I just asking you to align your code with the rest of `mmu.c`. If you refuse to do this, I will refrain from further review and ask you to post this patch directly into Zephyr mainline.

Reply:
how have you started this thread? did you first message contain enough information why have you requested this change?
why didn't you write it `I just asking you to align your code with the rest of mmu.c.`  at the beginning?
as far as I remember you said  that "I need to write good explanations why I want to see the change, because in other way author may behave in an angry way"

Reply:
This thread is sort of continuation of this: https://github.com/xen-troops/zephyr/pull/52#discussion_r1224563601

In that comment I wrote that we need to use `uintptr_t` because we are working with virtual addresses. I assumed that you'll infer that my first comment in this thread is related to the previous mention of `uintptr_t`.

Reply:
got it